src/hooks/twilio.ts

```diff
@@ -208,49 +206,51 @@ export const useVideoRoomState = ({
     hide: turnVideoOff,
   } = useShowHide(showVideoByDefault);
 
-  const localParticipant = room?.localParticipant;
-
-  useEffect(() => {
-    if (!localParticipant) return;
-
-    setParticipants((prevParticipants) => [
-      // Hopefully prevents duplicate users in the participant list
-      ...prevParticipants.filter(
-        (p) => p.identity !== localParticipant.identity
-      ),
-      localParticipant,
-    ]);
-  }, [localParticipant]);
-
-  useEffect(() => {
-    if (!token || !roomName) return;
-
-    const participantConnected = (participant: RemoteParticipant) => {
-      setParticipants((prevParticipants) => [...prevParticipants, participant]);
-    };
+  const participantConnected = useCallback((participant: RemoteParticipant) => {
+    setParticipants((prevParticipants) => [...prevParticipants, participant]);
+  }, []);
 
-    const participantDisconnected = (participant: RemoteParticipant) => {
+  const participantDisconnected = useCallback(
+    (participant: RemoteParticipant) => {
       setParticipants((prevParticipants) =>
         prevParticipants.filter((p) => p !== participant)
       );
-    };
+    },
+    []
+  );
+
+  useEffect(() => {
+    if (!token || !roomName) return;
 
     // https://media.twiliocdn.com/sdk/js/video/releases/2.7.1/docs/global.html#ConnectOptions
     connect(token, {
       name: roomName,
       video: hasVideo,
       enableDscp: true,
-    }).then((room) => {
-      setRoom(room);
-      room.on("participantConnected", participantConnected);
-      room.on("participantDisconnected", participantDisconnected);
-      room.participants.forEach(participantConnected);
-    });
+    }).then(setRoom);
 
     return () => {
       disconnect();
     };
-  }, [disconnect, roomName, token, hasVideo]);
+  }, [
+    disconnect,
+    roomName,
+    token,
+    hasVideo,
+    participantConnected,
+    participantDisconnected,
+  ]);
+
+  useEffect(() => {
+    if (!room) return;
+
+    room.on("participantConnected", participantConnected);
+    room.on("participantDisconnected", participantDisconnected);
+    setParticipants([
+      room?.localParticipant,
+      ...Array.from(room.participants.values()),
+    ]);
+  }, [room, participantConnected, participantDisconnected]);
```

Comment:
As called out in my comments on the previous review thread about this, we need to add the cleanup function to call `.off` to unregister these listeners. I explained this in more detail in that thread + added example code.

Reply:
I couldn't find the definitions for `.off` in https://media.twiliocdn.com/sdk/js/video/releases/2.9.0/docs/Room.html

Reply:
Added those + added a comment in https://github.com/sparkletown/sparkle/pull/1299/commits/ce562c6a4616118ea904bb8f088ca751787a5f95

Reply:
So twilio's documentation is just suboptimal and they do exist then? Your comment implies that you're unsure about it?